mongo/change_stream.go

```diff
@@ -386,8 +386,9 @@ func (cs *ChangeStream) buildPipelineSlice(pipeline interface{}) error {
 	cs.pipelineSlice = make([]bsoncore.Document, 0, val.Len()+1)
 
 	csIdx, csDoc := bsoncore.AppendDocumentStart(nil)
-	csDocTemp := cs.createPipelineOptionsDoc()
-	if cs.err != nil {
+
+	csDocTemp, err := cs.createPipelineOptionsDoc()
+	if err != nil {
 		return cs.err
```

Comment:
Same as above: It's confusing to check an error and then return a different error. Update this to return `err` (instead of `cs.err`) if it's not nil.